### ISSUE TITLE
Implement PCA analysis

### ISSUE THREAD

**Issue by danielmk** [barisckuru/phase-to-rate#8]
The PCA analysis is supposed to corroborate the perceptron analysis. The goal is to quantify how many PCAs are necessary to explain a fixed amount of variance. This should vary depending on distance and possibly on shuffling, because both change the amount of variance in the data. The sklearn implementation should be sufficient for our purposes: https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html

Specifically check the `explained_variance_` parameter.
